ghost/email-service/lib/BatchSendingService.js

```diff
@@ -360,14 +360,19 @@ class BatchSendingService {
 
         // Loop batches and send them via the EmailProvider
         let succeededCount = 0;
-        const queue = batches.slice();
+        const queue = batches.slice().map((batch, index) => {
+            return {batchIndex: index, ...batch};
+        });
```

Comment:
Since we use a `Promise.all()` to actually send all the batches with concurrency of 2 rather than a simple loop, we can't rely on any incremented counter within the loop to generate the delays — hence this initial loop to add the index to each batch, which is then used to calculate the `deliverytime` parameter below.

Reply:
Found a slightly cleaner way to do this I think, using `indexOf` rather than precalculating the index